crates/bevy_ecs/src/query/state.rs

```diff
@@ -96,6 +96,15 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
         let fetch_state = Q::init_state(world);
         let filter_state = F::init_state(world);
 
+        Self::new_with_state(world, fetch_state, filter_state)
+    }
+
+    /// Creates a new [`QueryState`] with custom initial state
```

Comment:
This could use a doc test; it's not going to be clear to users how or why to use this method.

Reply:
Done